### PR TITLE
`DualMatroid`: Add `is_valid` and `relabel` methods

### PR DESCRIPTION
**Pull request by gmou3**
Implement class-specific `is_valid` and `relabel` methods.
Previously, such functionality was achieved through the generic and very slow rank-based algorithms.

The PR also does a touch-up of the docstrings of `dual_matroid.py`.

### CODE REVIEW

File: src/sage/matroids/dual_matroid.py [hunk 1]
```diff
@@ -526,3 +502,64 @@ def __reduce__(self):
...
+        Return an isomorphic matroid with relabeled groundset.
+
+        The output is obtained by relabeling each element ``e`` by
+        ``mapping[e]``, where ``mapping`` is a given injective map. If
+        ``e not in mapping`` then the identity map is assumed.
+
+        INPUT:
+
+        - ``mapping`` -- a python object such that `mapping[e]` is the new
+          label of `e`
```
Comment (tscrim) [inline src/sage/matroids/dual_matroid.py]:
```suggestion
        - ``mapping`` -- a Python object such that ``mapping[e]`` is the new
          label of ``e``
```

**Review by tscrim — APPROVED** [review]
LGTM modulo one little thing.

File: src/sage/matroids/dual_matroid.py (hunk 1 above)
Comment (gmou3) [reply src/sage/matroids/dual_matroid.py]: I think that `e` is better than ``e`` (change it elsewhere too).

**Comment by tscrim** [comment]
LGTM. Thanks.

**Pull request merged**